lib/parsers/asm-parser-ewavr.ts

```diff
@@ -22,14 +22,44 @@
 // ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.
 
+import {ParsedAsmResult, ParsedAsmResultLine} from '../../types/asmresult/asmresult.interfaces';
+import {ParseFiltersAndOutputOptions} from '../../types/features/filters.interfaces';
+import {assert} from '../assert';
 import {logger} from '../logger';
+import {PropertyGetter} from '../properties.interfaces';
 import * as utils from '../utils';
 
 import {AsmParser} from './asm-parser';
 import {AsmRegex} from './asmregex';
 
+// this file uses null throughout for "not there" not undefined.
+type Source = {file: string | null; line: number};
+type Line = {text: string; source: Source | null};
+type Label = {
+    lines: Line[];
+    name: string | null;
+    initialLine: number;
+    file: string | undefined;
+    require: string[];
+};
+type ResultObject = {
+    prefix: Line[];
+    labels: Label[];
+    postfix: Line[];
+};
+
 export class AsmEWAVRParser extends AsmParser {
-    constructor(compilerProps) {
+    private readonly filenameComment: RegExp;
+    private readonly lineNumberComment: RegExp;
+    private readonly segmentBegin: RegExp;
+    private readonly segmentControl: RegExp;
+    private readonly definesLocal: RegExp;
+    private readonly miscDirective: RegExp;
+    private readonly dataStatement: RegExp;
+    private readonly requireStatement: RegExp;
+    private readonly beginFunctionMaybe: RegExp;
```

Comment:
Since these are just read-only constants defined in the constructor, they can be inlined into the property definition